components/net/fetch/request.rs

```diff
@@ -71,6 +84,7 @@ pub struct Request {
     pub preserve_content_codings: bool,
     // pub client: GlobalRef, // XXXManishearth copy over only the relevant fields of the global scope,
                               // not the entire scope to avoid the libscript dependency
+    pub is_service_worker_global_scope: Option<bool>,
```

Comment:
This isn't actually from the spec, it's a workaround for the client (refer to the note you left on the line above).